### PR TITLE
Initial Python 3.10 package forked from Python 3.9

### PR DESCRIPTION
**Pull request by hroncok**
Rawhide scratch build: https://koji.fedoraproject.org/koji/taskinfo?taskID=53074311

Main Python rawhide scratch build: https://koji.fedoraproject.org/koji/taskinfo?taskID=53074340

Once the [repo is created](https://pagure.io/releng/fedora-scm-requests/issue/29611) I'll open a giant PR for Fedora with this and the history to test it in Fedora CI and Zuul as well. Opened here for easier review of the changes.

Note that the pyc magic number is identical to 3.9, so marshalparser tests should pass (unless they are confused by the fact that this is called 3.10 but it quacks like 3.9).

I found out about a minor rpmlint warning when testing the rpmlintrc file, so I've opened https://src.fedoraproject.org/rpms/python3.9/pull-request/37 -- my intention is to rebase this on top of that later.

### CODE REVIEW

File: 00189-use-rpm-wheels.patch [hunk 1]
```diff
@@ -33,7 +33,7 @@ index 97dfa7ea71..984e587ea0 100644
  
 +_WHEEL_DIR = "/usr/share/python-wheels/"
  
--_SETUPTOOLS_VERSION = "49.2.1"
+-_SETUPTOOLS_VERSION = "47.1.0"
```
Comment (vstinner) [inline 00189-use-rpm-wheels.patch:36]: setuptools version is older in 3.10?

File: 00251-change-user-install-location.patch [hunk 2]
```diff
@@ -41,14 +41,15 @@ index aaa300efa9..f8d453912a 100644
...
+index 4d3b869fff..6c7b028f53 100644
 --- a/Lib/site.py
 +++ b/Lib/site.py
-@@ -353,7 +353,14 @@ def getsitepackages(prefixes=None):
+@@ -364,8 +364,14 @@ def getsitepackages(prefixes=None):
      return sitepackages
  
  def addsitepackages(known_paths, prefixes=None):
 -    """Add site-packages to sys.path"""
+-    _trace("Processing global site-packages")
```
Comment (vstinner) [inline 00251-change-user-install-location.patch:52]: Did you remove the _trace() call on purpose?

Comment (hroncok) [reply 00251-change-user-install-location.patch]: no, that was a bad rebase, fixing it

File: 00189-use-rpm-wheels.patch (hunk 1 above)
Comment (hroncok) [reply 00189-use-rpm-wheels.patch:36]: yes. I was surprised by this as well. the setuptools version was bumped between 3.9.0rc2 and 3.9.0 which obviously shocked me a bit.

Comment (hroncok) [reply 00189-use-rpm-wheels.patch:36]: This is tracked in https://github.com/python/cpython/pull/21748 and https://bugs.python.org/issue41490

File: 00251-change-user-install-location.patch (hunk 2 above)
Comment (vstinner) [reply 00251-change-user-install-location.patch]: Oh ok, thanks for the fix.

**Review by vstinner — APPROVED** [review]
LGTM.

I don't understand how the gpg comand uses the local pgp_keys.asc file, but I trust you on that part :-) Well, if the package builds successfully, I'm confident that this file is used, since I understood that Pablo's public key is not on the python.org list of GPG keys.

**Comment by hroncok** [comment]
> I don't understand how the gpg comand uses the local pgp_keys.asc file

    %gpgverify -k2 -s1 -d0

This translates to: Verify data from Source0 was signed into Source1 with key Source2.

**Comment by vstinner** [comment]
Oh! I didn't notice that it's a RPM macro and not a shell command. I very rarely use GPG. Thanks for the explanation ;-)

**Comment by frenzymadness** [comment]
The diff looks good to me. I'm looking forward to what the two digits in the version will break :smile: I'm gonna test Marshal parser with these builds.

I did a review and prepared the comment above and you've managed to force push to this PR before I've sent it. Should I take a look again?

**Comment by hroncok** [comment]
Not needed, it was just the rpmlint commit from python3.9.

**Pull request closed without merging**